app/controllers/issues_controller.rb

```diff
@@ -98,6 +112,24 @@ def update_status
     end
   end
 
+  # GET /issues/filter
+  def filter
+    @issues = @issues.where(status: params[:status]) if params[:status].present?
```

Comment:
this can be done in one line.
@issues = @issues.where(search_params)
def search_params
  params.permit(:status, :issue_type, :priority).delete_if { |_key, value| value.blank? }
end

Like query can be handled separately as you have done

Reply:
done